### ISSUE TITLE
Incorrect [SC2154: string1 is referenced but not assigned.]

### ISSUE THREAD

**Issue by Lathoudero** [koalaman/shellcheck#800]
#### For bugs
- SC2154
-"online" 
- [X ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [X ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash 
myFunction()
{
  eval string1="$1"
  eval string2="$2"
  eval string3="$3"

  echo "string1 = ${string1}"
  echo "string2 = ${string2}"
  echo "string3 = ${string3}"
}

var1="firstString"
var2="second string with spaces"
var3="thirdString"

myFunction "\${var1}" "\${var2}" "\${var3}"

exit 0

```

#### Here's what shellcheck currently says:
Line 8:
  echo "string1 = ${string1}"
                  ^-- SC2154: string1 is referenced but not assigned.
 
Line 9:
  echo "string2 = ${string2}"
                  ^-- SC2154: string2 is referenced but not assigned.
 
Line 10:
  echo "string3 = ${string3}"
                  ^-- SC2154: string3 is referenced but not assigned.

#### Here's what I wanted or expected to see:

$ shellcheck myscript
No issues detected!

If I don't use 'eval' shellcheck works fine. But In this case 'eval' is used to correctly refer to the variable.

**Comment by koalaman** [comment]
Closing for the same reason as #801. I'd love if this was generally fixable, but runtime dependent behavior is the bane static analysis.
